profiles/fscloud/variables.tf

```diff
@@ -0,0 +1,153 @@
+##############################################################################
+# Input Variables
+##############################################################################
+
+variable "resource_group_id" {
+  description = "ID of resource group to use when creating the mongodb database"
+  type        = string
+}
+
+variable "mongodb_version" {
+  description = "The version of mongodb. If null, the current default ICD mongodb version is used."
+  type        = string
+  default     = null
+}
+
+variable "plan" {
+  type        = string
+  description = "The name of the service plan that you choose for your mongodb instance"
+  default     = "standard"
+}
+
+variable "region" {
+  description = "The region mongodb is to be created on. The region must support BYOK ( us-south, us-east, and eu-de)"
+  type        = string
+  default     = "us-south"
+  validation {
+    condition = anytrue([
+      var.region == "us-south",
+      var.region == "us-east",
+      var.region == "eu-de"
+    ])
+    error_message = "region must be in a BYOK location, us-south, us-east or eu-de"
+  }
+}
+
+variable "allowlist" {
+  description = "(Optional, List of Objects) A list of allowed IP addresses for the database."
+  type = list(object({
+    address     = string
+    description = string
+  }))
+  default = []
+}
+
+variable "configuration" {
+  description = "Database Configuration in JSON format."
+  type = object({
+    maxmemory                   = optional(number)
+    maxmemory-policy            = optional(string)
+    appendonly                  = optional(string)
+    maxmemory-samples           = optional(number)
+    stop-writes-on-bgsave-error = optional(string)
+  })
+  default = null
+}
+
+variable "cpu_count" {
+  description = "Number of CPU cores available to the mongodb instance"
+  type        = number
+  default     = 7
+}
+
+variable "memory_mb" {
+  description = "Memory available to the mongodb instance"
+  type        = number
+  default     = 1024
+}
+
+variable "disk_mb" {
+  description = "Disk space available to the mongodb instance"
+  type        = number
+  default     = 20480
+}
+
+
+variable "instance_name" {
+  description = "Name of the new mongodb instance"
+  type        = string
+}
+
+variable "tags" {
+  type        = list(any)
+  description = "Optional, Array of Strings) A list of tags that you want to add to your instance."
+  default     = []
+}
+
+variable "key_protect_key_crn" {
```

Comment:
For FS cloud KYOK is the requirement. So this needs to be Hyper Protect Crypto Service (HPCS) crn and descriptions updated